garrysmod/gamemodes/terrortown/gamemode/vgui/sb_team.lua

```diff
@@ -116,11 +101,37 @@ end
 
 function PANEL:UpdateSortCache()
    self.rows_sorted = {}
-   for k,v in pairs(self.rows) do
-      table.insert(self.rows_sorted, v)
+
+   for _, row in pairs(self.rows) do
+      table.insert(self.rows_sorted, row)
    end
 
-   table.sort(self.rows_sorted, CompareScore)
+   table.sort(self.rows_sorted, function(rowa, rowb)
+      local plya = rowa:GetPlayer()
+      local plyb = rowb:GetPlayer()
+
+      if not IsValid(plya) then return false end
+      if not IsValid(plyb) then return true end
+
+      local comp = 0 -- Lua doesnt have an Ordering enumeration, I think?
+      
+      local sort_mode = GetConVar("ttt_scoreboard_sorting"):GetString()
+      local sort_func = sboard_panel.sort_table[sort_mode]
+
+      if sort_func != nil then
+         comp = sort_func(plya, plyb)
+      end
+
+      if comp != 0 then
+         return comp > 0
+      end
+
+      return plya:GetName() > plyb:GetName()

```

Comment:
This will sort "Z" before "a". You will have to upper or lowercase both strings to get a somewhat sane ordering.